backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/payment/paymentAPI.feature

```diff
@@ -0,0 +1,1439 @@
+      | inv_02022023_9          | bpartner_02022023_9      | bpartner_location_02022023_9      | 10 Tage 1 % | true      | CO        | true       |
+      | inv_02022023_10         | bpartner_02022023_9      | bpartner_location_02022023_9      | 10 Tage 1 % | true      | CO        | true       |
+
+  @from:cucumber
+  Scenario: Create OUTBOUND payment using the payment API
+  - fully allocate invoice's open amount with one JsonInvoicePaymentCreateRequest#line (no discount, no write off)
+  - use different currency and transaction date in the past
+
+    Given metasfresh contains M_PricingSystems
+      | Identifier                           | Name                                 | Value                                |
+      | paymentAllocPricingSystem_03022023_2 | paymentAllocPricingSystem_03022023_2 | paymentAllocPricingSystem_03022023_2 |
+
+    And metasfresh contains M_PriceLists
+      | Identifier                       | M_PricingSystem_ID.Identifier        | OPT.C_Country.CountryCode | C_Currency.ISO_Code | Name           | SOTrx | IsTaxIncluded | PricePrecision |
+      | paymentAllocPriceList_03022023_2 | paymentAllocPricingSystem_03022023_2 | DE                        | EUR                 | PriceListName1 | false | false         | 2              |
+    And metasfresh contains M_PriceList_Versions
+      | Identifier                 | M_PriceList_ID.Identifier        | ValidFrom  |
+      | paymentAllocPLV_03022023_2 | paymentAllocPriceList_03022023_2 | 2022-05-01 |
+
+    And metasfresh contains C_BPartners without locations:
+      | Identifier          | Name                    | Value                   | OPT.IsVendor | M_PricingSystem_ID.Identifier        | OPT.C_PaymentTerm_ID |
+      | bpartner_03022023_2 | BPartnerTest_03022023_2 | BPartnerTest_03022023_2 | Y            | paymentAllocPricingSystem_03022023_2 | 1000009              |
+
+    And metasfresh contains C_BPartner_Locations:
+      | Identifier                   | C_BPartner_ID.Identifier | OPT.IsShipToDefault | OPT.IsBillToDefault |
+      | bpartner_location_03022023_2 | bpartner_03022023_2      | Y                   | Y                   |
+
+    And load C_BP_BankAccount:
+      | C_BP_BankAccount_ID.Identifier | OPT.C_BP_BankAccount_ID |
+      | bp_bank_account_03022023_2     | 2000257                 |
+
+    And update C_BP_BankAccount:
+      | C_BP_BankAccount_ID.Identifier | OPT.C_Currency.ISO_Code |
+      | bp_bank_account_03022023_2     | CHF                     |
+
+    And load C_DocType:
+      | C_DocType_ID.Identifier | DocBaseType |
+      | outboundDocType         | APP         |
+
+    And load C_Currency:
+      | C_Currency_ID.Identifier | OPT.C_Currency_ID |
+      | eur                      | 102               |
+      | chf                      | 318               |
+
+    And load C_Conversion_Rate:
+      | C_Conversion_Rate_ID.Identifier | C_Currency_ID.Identifier | C_Currency_ID_To.Identifier | ValidFrom  | ValidTo    |
```

Comment:
i'm not sure what to make of these two conversion-rates.
- aren't both valid a the same time?
- which one is used by the conversion?

In my mind the tests should make sure that depending on the dates, in one case one conversion is used and in the other case the other conversion

Reply:
The conversion rate with identifier `currency_rate_1` is validFrom `2015-05-21` and the rate with identifier `currency_rate_2` is validFrom `2019-05-21`.
The `transactionDate` in the request is set to `2016-05-21` which means that `currency_rate_1` should be taken into consideration.
In the next step we validate the `DivideRate` for the two conversion rates, and those are:
 - `1.045236` for `currency_rate_1`
 - `1.13` for `currency_rate_2`
 
The `GrandTotal` of the invoice is `23.8 EUR`
The `amount` in the request is set to `24.88 CHF`
We validate that the invoice is paid in the end, which means that the `currency_rate_1` was taken into consideration (`24.88 CHF`/`1.045236` = `23.8 EUR`),
if it had been `currency_rate_2` the invoice would not have been paid (`24.88 CHF`/`1.13` = `22.01 EUR`).

That's how I was thinking about it.
Is it correct or am I missing something?

Reply:
hi @dm-ps yes - makes sense, thx for explaining the reasoning. 
what confused me was the `ValidTo` dates. I now checked the code (`CurrencyDAO#retrieveRateQuery`) and found that both `validFrom` and `validTo` are filtered by, and if more than one rate matches a date, then the one with the "youngest" (biggest) `validFrom` is used.
Please formulate this info and add it as a comment to the feature-file and the step-def. 
The goal being that i don't have to ask this again in 2 months :-)